src/java/org/apache/cassandra/locator/TokenMetadata.java

```diff
@@ -351,6 +370,8 @@ private void addBootstrapTokens(Collection<Token> tokens, InetAddress endpoint,
 
             for (Token token : tokens)
                 bootstrapTokens.put(token, endpoint);
+
+            logMultiMapDifference(String.format("bootstrapTokens changes for %s", original == null ? "bootstrap" : "node replacement"), bootstrapTokensSnapshot, bootstrapTokens.inverse());
```

Comment:
this is technically not a CompileTimeConstant, so we might have some issues marking these log messages as safe. Could you instead make "bootstrap" or "node replacement" a separate SafeArg that is logged?